### PR TITLE
Send notify message on last round

### PR DESCRIPTION
**Pull request by gavraz**
Send notification on last round instead of the end of the commit round.
This will make sure we use a different committee for the notify message.

### CODE REVIEW

**Comment by gavraz** [comment]
Automation tests passed.

**Comment by gavraz** [comment]
Please note this PR also fixes a bug: sending a notification message may be marked as sent even though it wasn't (eligibility for example).

File: hare/algorithm.go [hunk 1]
```diff
@@ -593,19 +603,12 @@ func (proc *ConsensusProcess) endOfRound3() {
 	}
 
 	// commit & send notification msg
```
Comment (barakshani) [inline hare/algorithm.go]: remove this

**Review by barakshani — APPROVED** [review]

File: hare/algorithm.go (hunk 1 above)
Comment (gavraz) [reply hare/algorithm.go]: Done

File: hare/algorithm.go [hunk 2]
```diff
@@ -592,20 +602,13 @@ func (proc *ConsensusProcess) endOfRound3() {
...
-		proc.Error("init default builder failed: %v", err)
-		return
-	}
-	builder = builder.SetType(Notify).SetCertificate(proc.certificate).Sign(proc.signing)
-	notifyMsg := builder.Build()
-	proc.sendMessage(notifyMsg)
-	proc.notifySent = true
+
+	// release proposal & commit trackers
+	proc.commitTracker = nil
```
Comment (gavraz) [inline hare/algorithm.go]: Add defer

**Review by zalmen — APPROVED** [review]

**Pull request merged**